doc/sphinx/language/gallina-specification-language.rst

```diff
@@ -835,27 +802,72 @@ Inductive definitions
 
 We gradually explain simple inductive types, simple annotated inductive
 types, simple parametric inductive types, mutually inductive types. We
-explain also co-inductive types.
+also explain co-inductive types.
 
 Simple inductive types
 ~~~~~~~~~~~~~~~~~~~~~~
 
-.. cmd:: Inductive @ident : {? @sort } := {? | } @ident : @type {* | @ident : @type }
+.. cmd:: Inductive @inductive_definition {* with @inductive_definition }
+
+   .. insertprodn inductive_definition record_binder_body
+
+   .. prodn::
+      inductive_definition ::= {? > } @ident_decl {* @binder } {? : @term } {? := {? @constructor_list_or_record_decl } } {? @decl_notations }
+      constructor_list_or_record_decl ::= %| {+| @constructor }
+      | @ident @constructor_type %| {*| @constructor }
+      | @ident @constructor_type
+      | @ident %{ {+; @record_field } %}
+      | %{ {+; @record_field } %}
+      constructor_type ::= {* @binder } {? @of_type }
+      record_field ::= {* #[ {*, @attr } ] } @record_binder {? %| @num } {? @decl_notations }
+      record_binder ::= @name
+      | @name @record_binder_body
+      record_binder_body ::= {* @binder } @of_type
+      | {* @binder } @of_type := @term
+      | {* @binder } := @term
 
    This command defines a simple inductive type and its constructors.
```

Comment:
Well, what used to be there for `cmd::` was incorrect, e.g. here's no @sort in the current grammar.

Probably what's needed here is an introductory taxonomy of inductive definition types and why/when you use each.  I understand mutually defined inductive types well enough to describe, but the others not so much.  Not entirely clear on which types/attributes are mutually exclusive. (e.g. can you have a mutually defined annotated inductive type?)  Seems like it would be a bit of a deep dive, so I would prefer to defer most of this to another PR.

Isn't the definition of "simple inductive type" more or less "any inductive type that doesn't use features A, B, C, ..."?  So you need to understand all of the alternatives up front.

Your thoughts?

Reply:
> there's no `@sort` in the current grammar.

There's `{? : @type }`. In the simple inductive case, `@type` is restricted to being a `@sort`.

>Not entirely clear on which types/attributes are mutually exclusive.

There are no mutual exclusive features. These really are more examples rather than a taxonomy. I.e. the original authors just wanted to introduce inductive types progressively rather than in full generality. We are taking the opposite path.

>Isn't the definition of "simple inductive type" more or less "any inductive type that doesn't use features A, B, C, ..."?

Right. So this is a good first example, but it shouldn't be defined negatively. It should just be presented as the simplest example of an inductive type.

Reply:
I was mistaken:`@sort` is defined in the grammar, though not visible in these productions.

The old cmd is `Inductive @ident : {? @sort } := {? | } @ident : @type {* | @ident : @type }`.  However, `: {? @sort }` should be `{? : @sort }`, then it should be simplified to:
`Inductive @ident {? : @sort } := {? | } {+| @ident : @type }`.  Also, the 

However, the relationship between this and the new syntax is not as obvious as I'd like:
- Using `@sort` in the old and `@type` in the new requires a sentence of explanation
- The old cmd drops `{? > } ... {* @binder } {? %| {* @binder } } ...  {? @decl_notations }` that are present in the new without explanation.
- The old syntax is very selective about what it includes from `constructors_or_record`
- `:=` is made unconditional while things in the old form remain conditional, such as `@sort`, which the text assumes is present.  Which should it be?

I suppose the text could say something like "Simple inductive types use a :n:`@type` that is a :n:`sort`.  <syntax notation> represents this in simplified form.", but yuck.  It would be a whole lot simpler just to begin with the `nat` example below and avoid showing a subset of the syntax in notation form at all.  (And probably avoid showing subset syntax anywhere in the doc.)

(IMHO this section needs a good bit more attention, but not in this PR.)

Reply:
I agree with your analysis.